### PR TITLE
[backport][depends] set CMAKE_SYSTEM_PROCESSOR on Toolchain_*.cmake.in files

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
Some addons and his dependencies built with cmake need the CMAKE_SYSTEM_PROCESSOR to prepare for the desired OS.
Further was this also not given to Kodi's own depends build

This is a backport from first commit of https://github.com/xbmc/xbmc/pull/17385

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: tools/depends/configure.ac [hunk 1]
```diff
@@ -325,7 +332,7 @@ case $host in
           if test "x$use_cpu" = "xauto"; then
             use_cpu=arm64
           fi
-
+          use_cmake_cpu=aarch64
```
Comment (AlwinEsch) [inline tools/depends/configure.ac]: Here a small difference to Kodi Matrix, it have `if test "x$use_cpu" = "xauto"; then` before on Matrix not.

File: tools/depends/configure.ac [hunk 2]
```diff
@@ -283,11 +288,13 @@ case $host in
             if test "x$use_cpu" = "xauto"; then
               use_cpu=x86_64
             fi
+            use_cmake_cpu=x86_64
           ;;
           i*86-apple-darwin*)
             if test "x$use_cpu" = "xauto"; then
               use_cpu=i386
             fi
+            use_cmake_cpu=i386
```
Comment (AlwinEsch) [inline tools/depends/configure.ac]: Also here a small difference, this is no more on Kodi Matrix present.

**Review by AlwinEsch — COMMENTED** [review]
Except two places the change is equal to Matrix.

**Comment by AlwinEsch** [comment]
@Rechi has cherry picked your commit about, is this OK for you?

**Comment by AlwinEsch** [comment]
jenkins build this with addons please

**Comment by DaveTBlake** [comment]
What PR to master does this this bacport @AlwinEsch ?

**Comment by AlwinEsch** [comment]
Is the first commit of https://github.com/xbmc/xbmc/pull/17385

**Comment by AlwinEsch** [comment]
Has used on begin a own way, see https://github.com/xbmc/xbmc/pull/17363. But them from @Rechi was easier and changed here from my to his way.

**Pull request merged**